pinot-common/src/main/java/com/linkedin/pinot/common/protocols/SegmentCompletionProtocol.java

```diff
@@ -384,7 +415,8 @@ public String toJsonString() {
       builder.append("{\"" + STATUS_KEY + "\":" + "\"" + _status.name() + "\"," + "\""
           + OFFSET_KEY + "\":" + _offset + ",\""
           + COMMIT_TYPE_KEY + "\":" + _isSplitCommit + ",\""
-          + SEGMENT_LOCATION_KEY + "\":\"" + _segmentLocation + "\"");
+          + SEGMENT_LOCATION_KEY + "\":\"" + _segmentLocation + "\"," + "\""
```

Comment:
you should add _segmentLocation and _controllerVipUrl only if they are non-null.